File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp" 
#include "example_interfaces/msg/string.hpp"

using namespace std::chrono_literals; // Use the std::chrono_literals namespace to allow for easy time duration literals

class RobotNewsStationNode : public rclcpp::Node 
{
    public:
        RobotNewsStationNode() : Node("robot_news_station"), robot_name_("C651")
        {
            publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news", 10);
            timer_ = this->create_wall_timer(0.5s, std::bind(&RobotNewsStationNode::publish_news, this));
            RCLCPP_INFO(this->get_logger(), "Here we go!");
        }
    private:
        void publish_news()
        {
            auto msg = example_interfaces::msg::String();
            msg.data = std::string("Hi this is " + robot_name_ + std::string("from Taiwan"));
            publisher_->publish(msg);
        }
        std::string robot_name_;
        rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
        rclcpp::TimerBase::SharedPtr timer_;
};


int main(int argc, char **argv) 
{
    rclcpp::init(argc, argv); 
    auto node = std::make_shared<RobotNewsStationNode>(); 
    rclcpp::spin(node); 
    rclcpp::shutdown(); 
    return 0; 
}

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3 
import rclpy 
from rclpy.node import Node 
from example_interfaces.msg import String

class RobotNewsStation(Node):
    def __init__(self):
        super().__init__("robot_news_station")
        self.declare_parameter("robot_name","C651")
        self.robot_name_ = self.get_parameter("robot_name").value
        # self.robot_name_ = "CCOP3587999"
        self.publishers_= self.create_publisher(String, "robot_news", 10) #robot_news is the topic name, 10 is the queue size
        self.timer_= self.create_timer(2, self.publish_news)
        self.get_logger().info("Robot news station is up and running!")

    def publish_news(self):
        msg=String()
        msg.data="Hi, this is " + self.robot_name_ + " reporting!"
        self.publishers_.publish(msg)

def main(args=None): 
    rclpy.init(args=args) 
    node = RobotNewsStation()  
    rclpy.spin(node)    
    rclpy.shutdown()

if __name__ == "__main__":  
    main()